.github/workflows/Benchmark.yml

```diff
@@ -16,13 +16,62 @@ on:
         description: url of baseline
 
 jobs:
+  Cache:
+    runs-on: ubuntu-latest
+    env:
+      TARGET_URL: ${{ github.event.inputs.target_url || format('{0}#{1}', github.event.pull_request.head.repo.html_url, github.event.pull_request.head.sha) }}
+      BASELINE_URL: ${{ github.event.inputs.baseline_url || format('{0}#{1}', github.event.pull_request.base.repo.html_url, github.event.pull_request.base.sha) }}
+    steps:
+      -
+        name: Print target&baseline repo url
+        run: |
+          echo "Target repo url: $TARGET_URL"
+          echo "Baseline repo url: $BASELINE_URL"
+      - uses: actions/checkout@v3
+      - id: restore-cache
+        uses: actions/cache/restore@v3
+        with:
+          path: ~/.julia
+          # FIXME: if branch updated frequently, old cache still exists, leading to old dependencies restored
+          # TODO: fake key, only want to use restore-keys, because now mainly using benchmark/Manifest.toml sha
+          #       to distinguish different commit in PR.
+          key: ${{ runner.os }}-bm-${{ env.TARGET_URL }}-${{ env.BASELINE_URL }}-
+          restore-keys: ${{ runner.os }}-bm-${{ env.TARGET_URL }}-${{ env.BASELINE_URL }}-
```

Comment:
Could we just add a date+time to the key? I don’t know if GHA has such a function built in.

Reply:
Adding "date+time" is a good idea that can make our keys unique. However, since our restoring and saving of cache are in two separate jobs, it may result in the saving job and the restoring job occurring at different times and causing cache misses.

Therefore, I suggest that we can use ${{ github.run_id }} and ${{ github. run_attempt }} to make our key unique (according to [Actions - Contexts](https://docs.github.com/en/actions/learn-github-actions/contexts), I can assure `run_id` and `attempt` collection can be used as the unique key).

But at the same time, we should consider that time can tell us when the cache was generated (which may be relevant for future debugging). Therefore, I think we should still include "date+time", but maybe place it behind the end of the key. (Actions Caches tends to show the entire contents of the cache name without omitting any part of it, so we can still see "date+time" easily)

How do you think about this idea

Reply:
Yeah that sounds like a good plan

Reply:
OK! Done in the new commit.